Source/WebCore/rendering/RenderProgress.cpp

```diff
@@ -81,27 +81,30 @@ bool RenderProgress::isDeterminate() const
 
 void RenderProgress::animationTimerFired()
 {
-    // FIXME: Progress bar animation should be performed as part of the rendering update
-    // lifecycle, to match the display's refresh rate.
+    // FIXME: ideally obtaining the repeat interval from Page is not RenderTheme-specific, but it
+    // currently is as it also determines whether we animate at all.
+    auto refreshInterval = theme().animationRepeatIntervalForProgressBar(*this);
+    if (!(refreshInterval > 0_s))
```

Comment:
The reason I added this is because refreshInterval could in theory become 0 due to the element or page disappearing or some such. And I don't think we want to have this code run in essentially a loop at that point.

Maybe you are correct though and `updateAnimationState()` would have been invoked in those cases.

@pxlcoder can you weigh in perhaps?

Reply:
If this is defensive coding for a case that we don’t know how to reproduce, then it might be safer to have it check after the `repaint` call, to guarantee we don’t leave something behind? Or maybe it’s better to check before like this does to avoid trying to do any work inside `repaint` that might lead to a crash. Or maybe `repaint` is guaranteed not to do anything in that case.

Beyond trying to write code that makes few assumptions, the best way to deal with a worry about the element or page going away is to think about how to construct a test. I hate to risk having such a consideration slow down an otherwise straightforward and important change, but I can’t help seeing the looseness of the idiom here.